### ISSUE TITLE
Feature Request: Physical interfaces on docker containers

### ISSUE THREAD

**Issue by satyajitbm** [docker/docker#25641]
I am using docker to bring up an application in a container. The application requires that the system have physical interfaces attached to it. Docker interfaces are virtual by default.

As a workaround, I followed the steps in [this post](http://jason.digitalinertia.net/exposing-docker-containers-with-sr-iov/) to add SR-IOV virtual functions of the host as interfaces on my containers using `pipework --direct-phys` flag. The problem with this is that I can only get a maximum of 64 virtual functions for a single physical interface(Intel 82599) in the hypervisor.

I understand containers are not meant to work like VMs, but is there a way to get physical interfaces in the containers before starting the container using `docker network` without mapping the interfaces in the host namespace? 

**Output of `docker version`:**

```
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:
 OS/Arch:      linux/amd64
```

**Output of `docker info`:**

```
Containers: 28
 Running: 28
 Paused: 0
 Stopped: 0
Images: 29
Server Version: 1.12.0
Storage Driver: devicemapper
 Pool Name: docker-253:0-711621-pool
 Pool Blocksize: 65.54 kB
 Base Device Size: 10.74 GB
 Backing Filesystem: xfs
 Data file: /dev/loop0
 Metadata file: /dev/loop1
 Data Space Used: 5.246 GB
 Data Space Total: 107.4 GB
 Data Space Available: 102.1 GB
 Metadata Space Used: 11.12 MB
 Metadata Space Total: 2.147 GB
 Metadata Space Available: 2.136 GB
 Thin Pool Minimum Free Space: 10.74 GB
 Udev Sync Supported: true
 Deferred Removal Enabled: false
 Deferred Deletion Enabled: false
 Deferred Deleted Device Count: 0
 Data loop file: /var/lib/docker/devicemapper/devicemapper/data
 WARNING: Usage of loopback devices is strongly discouraged for production use. Use `--storage-opt dm.thinpooldev` to specify a custom block storage device.
 Metadata loop file: /var/lib/docker/devicemapper/devicemapper/metadata
 Library Version: 1.02.107-RHEL7 (2015-10-14)
Logging Driver: json-file
Cgroup Driver: cgroupfs
Plugins:
 Volume: local
 Network: host overlay bridge null
Swarm: inactive
Runtimes: runc
Default Runtime: runc
Security Options: seccomp
Kernel Version: 3.10.0-229.20.1.el7.x86_64
Operating System: CentOS Linux 7 (Core)
OSType: linux
Architecture: x86_64
CPUs: 8
Total Memory: 62.74 GiB
Name: sim10.mvdce2esdvpn04.mv.nuagenetworks.net
ID: FKWY:YVCD:TTFN:A2YJ:EHF6:LP2L:GZBR:NIOU:PPRA:GLFO:W654:CM45
Docker Root Dir: /var/lib/docker
Debug Mode (client): false
Debug Mode (server): false
Registry: https://index.docker.io/v1/
WARNING: bridge-nf-call-iptables is disabled
WARNING: bridge-nf-call-ip6tables is disabled
Insecure Registries:
 127.0.0.0/8
```

**Additional environment details:**
I am running containers on a Centos 7.1.1503 bare metal machine.

**Steps to reproduce the issue:**
1. docker run -it centos:7 /bin/bash
2. ls -lrt /sys/class/net/

**Describe the results you received:**
\# ls -lrt /sys/class/net/
total 0
lrwxrwxrwx 1 root root 0 Aug 11 23:24 lo -> ../../devices/virtual/net/lo
lrwxrwxrwx 1 root root 0 Aug 11 23:24 eth0 -> ../../devices/virtual/net/eth0

**Describe the results you expected:**
\# ls -lrt /sys/class/net/
total 0
lrwxrwxrwx 1 root root 0 Aug  11 17:53 enp8s0 -> ../../devices/pci0000:00/0000:00:0a.0/0000:08:00.0/net/enp8s0

A workaround where the bottleneck is not on the number of physical interfaces also works.

**Comment by justincormack** [comment]
The Intel 82599 does gave that limit of SR-IOV devices, 63 per physical
port. Dual port devices have twice as many. I believe the 100Gb devices may
have more. These devices were designed for VM uses cases, hence the limits.

If you need physical devices this us the best option still I believe.
